tests/patmat/i4225.scala

```diff
@@ -0,0 +1,12 @@
+object Bar {
+  def unapply(x: Int): Some[Int] =
+    Some(0)
+}
+
+object Test {
+  def test(x: Int) =
+    x match {
+      case Bar(a) => a
+      case _ => x // should be unreachable
```

Comment:
@allanrenucci : the exhaustivity check both in Dotty and Scalac ignores `null`. Otherwise, all patterns without a `_` will not be exhaustive. Thus in the example above, both Dotty & Scalac will report a warning.

Reply:
Sorry, I was wrong, for reachability check we can handle `null` thus avoid the warning:

It's implemented in https://github.com/lampepfl/dotty/compare/master...dotty-staging:fix-4225?expand=1#diff-5353f08e8d531fe0ee59335438e6cf7a.